src/mds/CInode.cc

```diff
@@ -434,64 +435,98 @@ sr_t *CInode::project_snaprealm(snapid_t snapid)
     new_srnode->created = snapid;
     new_srnode->current_parent_since = get_oldest_snap();
   }
+  return new_srnode;
+}
+
+void CInode::project_snaprealm(sr_t *new_srnode)
+{
   dout(10) << "project_snaprealm " << new_srnode << dendl;
+  assert(projected_nodes.back()->snapnode == projected_inode_t::UNDEF_SRNODE);
   projected_nodes.back()->snapnode = new_srnode;
   ++num_projected_srnodes;
-  return new_srnode;
 }
 
-/* if newparent != parent, add parent to past_parents
- if parent DNE, we need to find what the parent actually is and fill that in */
 void CInode::project_past_snaprealm_parent(SnapRealm *newparent)
 {
   sr_t *new_snap = project_snaprealm();
+  record_past_snaprealm_parent(new_snap, newparent);
+}
+
+
+/* if newparent != parent, add parent to past_parents
+ if parent DNE, we need to find what the parent actually is and fill that in */
+void CInode::record_past_snaprealm_parent(sr_t *new_snap, SnapRealm *newparent)
+{
   SnapRealm *oldparent;
   if (!snaprealm) {
     oldparent = find_snaprealm();
     new_snap->seq = oldparent->get_newest_seq();
-  }
-  else
+  } else {
     oldparent = snaprealm->parent;
+  }
 
   if (newparent != oldparent) {
+    // convert past_parents to past_parent_snaps
```

Comment:
@amitkumar50 
Eh, That I don't get: No Space between `//` and Convert.
I would render comments illegible.